src/Network/Cookie/RequestCookieJar.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Cake\Network\Cookie;
+
+class RequestCookieJar
+{
+    protected $_cookies = [];
+
+    public function __construct()
+    {
+        foreach ($_COOKIE as $name => $value) {
```

Comment:
I would take the cookies as a parameter. Right now there is no way for a fake request to easily pass cookies in. In general I've been trying to herd superglobal references into as few places as possible.